### ISSUE TITLE
The quantization scripts of TorchVision no longer work using the latest nightly of Core 

### ISSUE THREAD

**Issue by z-a-f** [pytorch/pytorch#69235]
It looks like torchvision is broken due to some older changes in the quant. Here is the original issue from here https://github.com/pytorch/vision/pull/4997#discussion_r757569785_

It seems that the situation is much more broken than we thought. The quantization scripts of TorchVision no longer work using the latest nightly of Core (`pytorch                   1.11.0.dev20211125 py3.9_cuda11.1_cudnn8.0.5_0    pytorch-nightly`):
```
$ python train_quantization.py --device='cpu' --post-training-quantize --backend='fbgemm' --model='shufflenet_v2_x0_5'
Not using distributed mode
Namespace(data_path='/datasets01/imagenet_full_size/061417/', model='shufflenet_v2_x0_5', backend='fbgemm', device='cpu', batch_size=32, eval_batch_size=128, epochs=90, num_observer_update_epochs=4, num_batch_norm_update_epochs=3, num_calibration_batches=32, workers=16, lr=0.0001, momentum=0.9, weight_decay=0.0001, lr_step_size=30, lr_gamma=0.1, print_freq=10, output_dir='.', resume='', start_epoch=0, cache_dataset=False, sync_bn=False, test_only=False, post_training_quantize=True, world_size=1, dist_url='env://', interpolation='bilinear', val_resize_size=256, val_crop_size=224, train_crop_size=224, clip_grad_norm=None, weights=None, distributed=False)
Loading data
Loading data
Loading training data
Took 5.208142042160034
Loading validation data
Creating data loaders
Creating model shufflenet_v2_x0_5
./torch/ao/quantization/observer.py:177: UserWarning: Please use quant_min and quant_max to specify the range for observers.                     reduce_range will be deprecated in a future release of PyTorch.
  warnings.warn(
Calibrating
Traceback (most recent call last):
  File "./vision/references/classification/train_quantization.py", line 274, in <module>
    main(args)
  File "./vision/references/classification/train_quantization.py", line 103, in main
    evaluate(model, criterion, data_loader_calibration, device=device, print_freq=1)
  File "./vision/references/classification/train.py", line 76, in evaluate
    output = model(image)
  File "./torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "./vision/torchvision/models/quantization/shufflenetv2.py", line 49, in forward
    x = self._forward_impl(x)
  File "./vision/torchvision/models/shufflenetv2.py", line 145, in _forward_impl
    x = self.conv1(x)
  File "./torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "./torch/nn/modules/container.py", line 141, in forward
    input = module(input)
  File "./torch/nn/modules/module.py", line 1131, in _call_impl
    hook_result = hook(self, input, result)
  File "./torch/ao/quantization/quantize.py", line 91, in _observer_forward_hook
    return self.activation_post_process(output)
  File "./torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "./torch/ao/quantization/observer.py", line 1093, in forward
    combined_histogram = torch.histc(
RuntimeError: Inplace update to inference tensor outside InferenceMode is not allowed.You can make a clone to get a normal tensor before doing inplace update.See https://github.com/pytorch/rfcs/pull/17 for more details.
```

I have confirmed that there are no inplace operations on the network (all of them are replaced prior quantizing).

Installing `pytorch                   1.11.0.dev20211020 py3.9_cuda11.1_cudnn8.0.5_0    pytorch-nightly` and rerunning the command works. All experiments were run using the latest nightly for TorchVision.

I also tried recent nightlies from the last week or so, but doesn't work. Something broke it earlier this month. 

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo

_Originally posted by @datumbox in https://github.com/pytorch/vision/pull/4997#discussion_r757569785_

**Comment by z-a-f** [comment]
@datumbox if  (by any chance0)you have a bisect for pytorch, please post it here. Otherwise, I wlll run it later today

**Comment by z-a-f** [comment]
> It seems that the situation is much more broken than we thought. The quantization scripts of TorchVision no longer work using the latest nightly of Core (`pytorch 1.11.0.dev20211125 py3.9_cuda11.1_cudnn8.0.5_0 pytorch-nightly`):
> 
> ```
> $ python train_quantization.py --device='cpu' --post-training-quantize --backend='fbgemm' --model='shufflenet_v2_x0_5'
> Not using distributed mode
> Namespace(data_path='/datasets01/imagenet_full_size/061417/', model='shufflenet_v2_x0_5', backend='fbgemm', device='cpu', batch_size=32, eval_batch_size=128, epochs=90, num_observer_update_epochs=4, num_batch_norm_update_epochs=3, num_calibration_batches=32, workers=16, lr=0.0001, momentum=0.9, weight_decay=0.0001, lr_step_size=30, lr_gamma=0.1, print_freq=10, output_dir='.', resume='', start_epoch=0, cache_dataset=False, sync_bn=False, test_only=False, post_training_quantize=True, world_size=1, dist_url='env://', interpolation='bilinear', val_resize_size=256, val_crop_size=224, train_crop_size=224, clip_grad_norm=None, weights=None, distributed=False)
> Loading data
> Loading data
> Loading training data
> Took 5.208142042160034
> Loading validation data
> Creating data loaders
> Creating model shufflenet_v2_x0_5
> ./torch/ao/quantization/observer.py:177: UserWarning: Please use quant_min and quant_max to specify the range for observers.                     reduce_range will be deprecated in a future release of PyTorch.
>   warnings.warn(
> Calibrating
> Traceback (most recent call last):
>   File "./vision/references/classification/train_quantization.py", line 274, in <module>
>     main(args)
>   File "./vision/references/classification/train_quantization.py", line 103, in main
>     evaluate(model, criterion, data_loader_calibration, device=device, print_freq=1)
>   File "./vision/references/classification/train.py", line 76, in evaluate
>     output = model(image)
>   File "./torch/nn/modules/module.py", line 1110, in _call_impl
>     return forward_call(*input, **kwargs)
>   File "./vision/torchvision/models/quantization/shufflenetv2.py", line 49, in forward
>     x = self._forward_impl(x)
>   File "./vision/torchvision/models/shufflenetv2.py", line 145, in _forward_impl
>     x = self.conv1(x)
>   File "./torch/nn/modules/module.py", line 1110, in _call_impl
>     return forward_call(*input, **kwargs)
>   File "./torch/nn/modules/container.py", line 141, in forward
>     input = module(input)
>   File "./torch/nn/modules/module.py", line 1131, in _call_impl
>     hook_result = hook(self, input, result)
>   File "./torch/ao/quantization/quantize.py", line 91, in _observer_forward_hook
>     return self.activation_post_process(output)
>   File "./torch/nn/modules/module.py", line 1110, in _call_impl
>     return forward_call(*input, **kwargs)
>   File "./torch/ao/quantization/observer.py", line 1093, in forward
>     combined_histogram = torch.histc(
> RuntimeError: Inplace update to inference tensor outside InferenceMode is not allowed.You can make a clone to get a normal tensor before doing inplace update.See https://github.com/pytorch/rfcs/pull/17 for more details.
> ```
> 
> I have confirmed that there are no inplace operations on the network (all of them are replaced prior quantizing).
> 
> Installing `pytorch 1.11.0.dev20211020 py3.9_cuda11.1_cudnn8.0.5_0 pytorch-nightly` and rerunning the command works. All experiments were run using the latest nightly for TorchVision.
> 
> I also tried recent nightlies from the last week or so, but doesn't work. Something broke it earlier this month.
> 
> cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo
> 
> _Originally posted by @datumbox in [pytorch/vision#4997 (comment)](https://github.com/pytorch/vision/pull/4997#discussion_r757569785)_

**Comment by datumbox** [comment]
@z-a-f Unfortunately I don't have it. I just know that one working version was pytorch 1.11.0.dev20211020.

**Comment by andrewor14** [comment]
I was able to confirm that #68143 did not break this. However, it did directly cause the other issue (extra items in state_dict) that https://github.com/pytorch/vision/pull/4997 fixes. We are discussing a way to fix it.

**Comment by datumbox** [comment]
@z-a-f @andrewor14 The issue is fixed on the latest core. It's unclear which PR did. I closed the issue on TorchVision side and I can keep an eye in case this pops up again. Shall we close this ticket?

**Comment by andrewor14** [comment]
Yes, in case we want to investigate/bisect in the future, this issue was fixed after [bcd0303](https://github.com/pytorch/pytorch/commit/bcd0303) but some time before 12/9. Closing for now. Thanks!
